tests/migration/postcopy.go

```diff
@@ -0,0 +1,412 @@
+		libvmi.WithServiceAccountDisk("default"),
+	)
+
+	mode := v1.MigrationPreCopy
+	if migrationPolicy != nil && migrationPolicy.Spec.AllowPostCopy != nil && *migrationPolicy.Spec.AllowPostCopy {
+		mode = v1.MigrationPostCopy
+	}
+
+	// postcopy needs a privileged namespace
+	if mode == v1.MigrationPostCopy {
+		vmi.Namespace = testsuite.NamespacePrivileged
+	}
+
+	disks := vmi.Spec.Domain.Devices.Disks
+	disks[len(disks)-1].Serial = secretDiskSerial
+
+	if migrationPolicy != nil {
+		AlignPolicyAndVmi(vmi, migrationPolicy)
+		migrationPolicy = CreateMigrationPolicy(virtClient, migrationPolicy)
+	}
+	vmi = tests.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 180)
+
+	// Wait for cloud init to finish and start the agent inside the vmi.
+	Eventually(matcher.ThisVMI(vmi), 12*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
+
+	By("Checking that the VirtualMachineInstance console has expected output")
+	Expect(console.LoginToFedora(vmi)).To(Succeed(), "Should be able to login to the Fedora VM")
+
+	if mode == v1.MigrationPostCopy {
+		By("Running stress test to allow transition to post-copy")
+		runStressTest(vmi, stressLargeVMSize, stressDefaultSleepDuration)
+	}
+
+	// execute a migration, wait for finalized state
+	By("Starting the Migration for iteration")
+	sourcePod, err := libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+	Expect(err).NotTo(HaveOccurred())
+
+	migration := libmigration.New(vmi.Name, vmi.Namespace)
+	migration = libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(virtClient, migration)
+	By("Checking VMI, confirm migration state")
+	vmi = libmigration.ConfirmVMIPostMigration(virtClient, vmi, migration)
+	Expect(vmi.Status.MigrationState.SourcePod).To(Equal(sourcePod.Name))
+	confirmMigrationMode(virtClient, vmi, mode)
+
+	By("Is agent connected after migration")
+	Eventually(matcher.ThisVMI(vmi), 12*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
+
+	By("Checking that the migrated VirtualMachineInstance console has expected output")
+	Expect(console.OnPrivilegedPrompt(vmi, 60)).To(BeTrue(), "Should stay logged in to the migrated VM")
+
+	By("Checking that the service account is mounted")
+	Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+		&expect.BSnd{S: "cat /mnt/servacc/namespace\n"},
+		&expect.BExp{R: vmi.Namespace},
+	}, 30)).To(Succeed(), "Should be able to access the mounted service account file")
+}
+
+func confirmMigrationMode(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, expectedMode v1.MigrationMode) {
+	By("Retrieving the VMI post migration")
+	vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, &metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Verifying the VMI's migration mode")
+	Expect(vmi.Status.MigrationState.Mode).To(Equal(expectedMode))
+}
```

Comment:
Can we move this under `libmigration` pkg and remove the duplicate in the swap_test? I know it's outside of this PR :(